CRM/Contribute/BAO/Contribution.php

```diff
@@ -3447,6 +3456,12 @@ public static function updateFinancialAccounts(&$params, $context = NULL, $skipT
     }
     if ($context == 'changedAmount' || $context == 'changeFinancialType') {
       $itemAmount = $params['trxnParams']['total_amount'] = $params['trxnParams']['net_amount'] = $params['total_amount'] - $params['prevContribution']->total_amount;
+      if (isset($params['prevContribution']->tax_amount)) {
```

Comment:
I am not sure about the logic here since I don't have enough context .. but shouldn't this be :

```php
if (isset($params['tax_amount'])) {
```
instead